labgrid/factory.py

```diff
@@ -80,6 +80,23 @@ def _convert_to_named_list(self, data):
             assert 'cls' in item
         return result
 
+    def normalize_role(self, name, config):
+        role = config.get('role', name)
```

Comment:
This function is named `normalize_role` but does not do anything with the role. Rename to `normalize_config` and remove the role parameter?

Reply:
Renamed to normalize_config and dropped the name parameter.